nixos/modules/services/misc/ollama.nix

```diff
@@ -1,75 +1,76 @@
 { config, lib, pkgs, ... }:
 let
-  inherit (lib) types mkBefore;
+  inherit (lib) literalExpression types mkBefore;
 
   cfg = config.services.ollama;
   ollamaPackage = cfg.package.override {
     inherit (cfg) acceleration;
   };
+
+  staticUser = cfg.user != null && cfg.group != null;
 in
 {
   imports = [
     (lib.mkRemovedOptionModule [ "services" "ollama" "listenAddress" ]
       "Use `services.ollama.host` and `services.ollama.port` instead.")
+    (lib.mkRemovedOptionModule [ "services" "ollama" "sandbox" ]
+      "Set `services.ollama.user` and `services.ollama.group` instead.")
+    (lib.mkRemovedOptionModule [ "services" "ollama" "writablePaths" ]
+      "The `models` directory is now always writable. To make other directories writable, use `systemd.services.ollama.serviceConfig.ReadWritePaths`." )
   ];
 
   options = {
     services.ollama = {
       enable = lib.mkEnableOption "ollama server for local large language models";
       package = lib.mkPackageOption pkgs "ollama" { };
+
+      user = lib.mkOption {
+        type = with types; nullOr str;
+        default = null;
+        example = "ollama";
+        description = ''
+          User account under which to run ollama. Defaults to [`DynamicUser`](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=)
+          when set to `null`.
+
+          The user will automatically be created, if this option is set to a non-null value.
+        '';
+      };
+
+      group = lib.mkOption {
+        type = with types; nullOr str;
+        default = cfg.user;
+        defaultText = literalExpression "services.ollama.user";
+        example = "ollama";
+        description = ''
+          Group under which to run ollama. Only used when `services.ollama.user` is set.
+
+          The group will automatically be created, if this option is set to a non-null value.
+        '';
+      };
+
       home = lib.mkOption {
         type = types.str;
-        default = "%S/ollama";
+        default = "/var/lib/ollama";
         example = "/home/foo";
         description = ''
           The home directory that the ollama service is started in.
-
-          See also `services.ollama.writablePaths` and `services.ollama.sandbox`.
         '';
       };
+
       models = lib.mkOption {
         type = types.str;
-        default = "%S/ollama/models";
+        default = "${cfg.home}/models";
+        defaultText = literalExpression "\${config.ervices.ollama.home}/models";
```

Comment:
Typo: `services` is missing an `s`.

Also, I'm don't think that `literalExpression` is a good fit here, since the value is indeed an interpolated string, not a bare path. I'm pretty sure the `${}` interpolation syntax doesn't even work outside of strings.

Reply:
If you prefer, you could also change it to something like:
``` nix
literalExpression ''config.services.ollama.home + "/models"''
```

Reply:
There is nothing wrong with interpolation, given that the option uses `type = str`, so unless we are going for `type = path` the change seems not that important.

Reply:
`ervices` is a typo and you’re missing the internal quotes (`''"…"''`).

Reply:
Indeed, I prefer showing it as interpolation, I was simply expanding on potential options.

Reply:
@emilazy  As far as I understand, `literalExpression` simply renders strings as-is, but without the quotes that it adds by default. So using `literalExpression` with internal quotes is simply doing the same thing as not using `literalExpression` at all, but with extra steps (added complexity and cognitive overhead).

Reply:
Oh, true. Never mind me then. I was mixing it up with `literalMD` or whatever it's called now.